### PR TITLE
Create rest api client library

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

- Create rest api client library.
- Moved Rest API Request and Response typings to service library package 
- Authentication is not included in this PR.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/web-api-client-library/src/a11y-service-client.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    private getRequestOptions(requestBody?: Object): request.RequestPromiseOptions {
+        return {
+            json: requestBody,
+            qs: {
+                'api-version': this.apiVersion,
+            },
+            headers: {
+                'Content-Type': 'application/json',
```
Comment (devaradhanm) [inline packages/web-api-client-library/src/a11y-service-client.ts]: you might also need to pass auth token that you get by calling aad

File: packages/web-api-client-library/src/a11y-service-client.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { injectable } from 'inversify';
+import * as request from 'request-promise';
+
+@injectable()
+export class A11yServiceClient {
+    constructor(private readonly baseUrl: string, private readonly apiVersion = '1.0', private readonly httpRequest = request) {}
```
Comment (devaradhanm) [inline packages/web-api-client-library/src/a11y-service-client.ts]: how do you make sure that you don't create more than one http agent?

File: .vscode/workspace.code-workspace [hunk 3]
```diff
@@ -64,6 +64,10 @@
             "name": "web-workers",
             "path": "../packages/web-workers"
         },
+        {
+            "name": "web-api-client-library",
```
Comment (devaradhanm) [inline .vscode/workspace.code-workspace]: may be just call it web-api-client

Comment (haonliu) [reply .vscode/workspace.code-workspace]: updated.

File: packages/web-api-client-library/src/a11y-service-client.ts (hunk 1 above)
Comment (haonliu) [reply packages/web-api-client-library/src/a11y-service-client.ts]: As discussed offline auth will not be included in this PR.

File: packages/web-api-client-library/src/a11y-service-client.ts (hunk 2 above)
Comment (haonliu) [reply packages/web-api-client-library/src/a11y-service-client.ts]: We are using node js request package which provides option to use [forever-agent](https://github.com/request/forever-agent).

Comment (devaradhanm) [reply packages/web-api-client-library/src/a11y-service-client.ts]: does it create a new forever agent for every request, or does it reuse the same agent?

Comment (haonliu) [reply packages/web-api-client-library/src/a11y-service-client.ts]: discussed offline

Comment (haonliu) [reply packages/web-api-client-library/src/a11y-service-client.ts]: Verified that the client will reuse the same https agent.

File: packages/web-api-client/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "web-api-client-library",
```
Comment (devaradhanm) [inline packages/web-api-client/package.json]: call it web-api-client

File: packages/web-api-client/src/register-web-api-client-library-to-container.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as inversify from 'inversify';
+import { A11yServiceClient } from './a11y-service-client';
+
+export function registerWebApiClientLibraryToContainer(container: inversify.Container): inversify.Container {
```
Comment (devaradhanm) [inline packages/web-api-client/src/register-web-api-client-library-to-container.ts]: this should not be in this package. let this be inside the web-worker's container setup.

File: packages/web-api-client/package.json (hunk 4 above)
Comment (haonliu) [reply packages/web-api-client/package.json]: updated

**Pull request merged**